x-pack/plugins/spaces/server/default_space/default_space_service.ts

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { CoreSetup, SavedObjectsServiceStart, Logger, ServiceStatus } from 'src/core/server';
+import {
+  concat,
+  of,
+  timer,
+  Observable,
+  ObservableInput,
+  combineLatest,
+  defer,
+  Subscription,
+  BehaviorSubject,
+} from 'rxjs';
+import { mergeMap, switchMap, catchError, tap } from 'rxjs/operators';
+import { ServiceStatusLevels } from '../../../../../src/core/server';
+import { ILicense } from '../../../licensing/server';
+import { SpacesLicense } from '../../common/licensing';
+import { createDefaultSpace } from './create_default_space';
+
+interface Deps {
+  coreStatus: CoreSetup['status'];
+  getSavedObjects: () => Promise<Pick<SavedObjectsServiceStart, 'createInternalRepository'>>;
+  license$: Observable<ILicense>;
+  spacesLicense: SpacesLicense;
+  logger: Logger;
+}
+
+export const RETRY_SCALE_DURATION = 100;
+export const RETRY_DURATION_MAX = 10000;
+
+const calculateDuration = (i: number) => {
+  const duration = i * RETRY_SCALE_DURATION;
+  if (duration > RETRY_DURATION_MAX) {
+    return RETRY_DURATION_MAX;
+  }
+
+  return duration;
+};
+
+// we can't use a retryWhen here, because we want to propagate the unavailable status and then retry
+const propagateUnavailableStatusAndScaleRetry = () => {
+  let i = 0;
+  return (err: Error, caught: ObservableInput<any>) =>
+    concat(
+      of({
+        level: ServiceStatusLevels.unavailable,
+        summary: `Error creating default space: ${err.message}`,
+      }),
+      timer(calculateDuration(++i)).pipe(mergeMap(() => caught))
+    );
+};
+
+export class DefaultSpaceService {
+  private initializeSubscription?: Subscription;
+
+  private serviceStatus$?: BehaviorSubject<ServiceStatus>;
+
+  public setup({ coreStatus, getSavedObjects, license$, spacesLicense, logger }: Deps) {
+    const statusLogger = logger.get('status');
+
+    this.serviceStatus$ = new BehaviorSubject({
+      level: ServiceStatusLevels.unavailable,
+      summary: 'not initialized',
+    } as ServiceStatus);
+
+    this.initializeSubscription = combineLatest([coreStatus.core$, license$])
```

Comment:
This `rxjs` flow was taken/inspired by the legacy [`watch_status_and_license_to_initialize.js`](https://github.com/elastic/kibana/blob/8e9a8a84dccfa7965ce8a22362885e6cdef8b51f/x-pack/legacy/server/lib/watch_status_and_license_to_initialize.js) utility